src/lib/chat/matrix-client.ts

```diff
@@ -281,16 +354,25 @@ export class MatrixClient implements IChatClient {
     this.connectionStatus = connectionStatus;
   }
 
+  processMessageEvent(event): void {
+    if (event.type === EventType.RoomMessage) {
+      const relatesTo = event.content[MatrixClient.RELATES_TO];
+      if (relatesTo && relatesTo.rel_type === MatrixClient.REPLACE) {
+        this.onMessageUpdated(event);
+      } else {
+        this.publishMessageEvent(event);
+      }
+    }
+  }
+
   private async initializeEventHandlers() {
     this.matrix.on('event' as any, async ({ event }) => {
       console.log('event: ', event);
       if (event.type === EventType.RoomEncryption) {
         console.log('encryped message: ', event);
       }
 
-      if (event.type === EventType.RoomMessage) {
-        this.publishMessageEvent(event);
-      }
+      this.processMessageEvent(event);
```

Comment:
Did you test this in old unencrypted rooms or in an encrypted one? I think there's a decrypted event that needs to handle this :thinking:

Reply:
Correct, thanks for pointing that out. I reused the `processMessageEvent` to handle this.